### PR TITLE
[SPARK-29596][Web UI] Task duration not updating for running tasks

### PR DESCRIPTION
**Pull request by Le-Dai**
### What changes were proposed in this pull request?

Changed the source of web ui duration information
### Why are the changes needed?

Task duration not updating for running tasks

### Does this PR introduce any user-facing change?

no

### How was this patch tested?

cat web ui

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala [hunk 1]
```diff
@@ -848,7 +848,7 @@ private[ui] class TaskPagedTable(
         </div>
       </td>
       <td>{UIUtils.formatDate(task.launchTime)}</td>
-      <td>{formatDuration(task.taskMetrics.map(_.executorRunTime))}</td>
```
Comment (Ngone51) [inline core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala]: Hi @MrDLontheway , please note that this change is intentionally introduced in #23081.

**Review by Ngone51 — COMMENTED** [review]
Would you please explain more about why task duration doesn't update in UI?

Besides, does this problem only exists in branch-2.4? Because I also see the same code in master branch.

**Comment by MrDLontheway** [comment]
> Would you please explain more about why task duration doesn't update in UI?
> 
> Besides, does this problem only exists in branch-2.4? Because I also see the same code in master branch.

Thank you for your reminder, during the task running process, because the executive run time is not updated, the duration will not be updated. I will seek other solutions

**Comment by Ngone51** [comment]
> because the executive run time is not updated, the duration will not be updated.

I know that executor run time doesn't update according to your description, but I also want to know the root cause why it doesn't change.

**Comment by gengliangwang** [comment]
@MrDLontheway could you provide more details and screenshot for the issue?
Also, PR should be created against the master branch.

**Comment by MrDLontheway** [comment]
ui
when task in RUNNING status , duration not update.
3.0.0 Removed the detail list and changed the duration calculation method, I don't know if it should be submitted to the master branch

**Review by planga82 — COMMENTED** [review]
I'm not sure about this change. There are a lot of more information that is used from metrics to create TaskDataWrapper, so, the other metrics are not updated? If this is the case the problem could be we are not updating metrics in the right way and we have to change it and not this specific metric only.

**Comment by holdenk** [comment]
Jenkins ok to test

**Pull request closed without merging**